### ISSUE TITLE
rename FunctionMaxLength to FunctionNameMaxLength

### ISSUE THREAD

**Issue by robstoll** [detekt/detekt#2403]
## Context
IMO the name is a bit misleading, I figured first it is concerning the length of the functions body. 
The description mentions name as well so I think it would be better to rename this rule
https://github.com/arturbosch/detekt/blob/master/detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/FunctionMaxLength.kt

**Comment by schalkms** [comment]
The mentioned rule is located in the `naming` ruleset, so I think the name fits.
Furthermore, there are other rules with the same naming scheme in the same ruleset (for instance VariableMaxLength & VariableMinLength). Hence, the naming scheme is actually consistent here.

I hope to have a stable ruleset with long-term support. As a user I'd also expect this behavior.
This raises the barrier for detekt's users to update older detekt versions.
For me it feels like making a change for the sake of changing.
If we start doing this, this will cause trouble in the future.
I'm highly against doing this and hope that you can rethink this decision @arturbosch .

I agree that we can change rule names if the behavior of the rule changes.
Moreover, we should also have a second thought about the naming when introducing new rules to detekt in order to be consistent.

**Comment by robstoll** [comment]
I see your point but...
```
@Suppress("FunctionMaxLength")
fun myLovelyFunctionWhichComputesAnImportantFigure() {
  // some
  // lines
  // but
  // not 
  // as 
  // many
  // as 
  // it
  // causes
  // a
  // detekt 
  // issue
}
```
Looks to me like one suppressed the functions length and not the name. 
Thus, either:
- enable that one can write `Suppress("naming.FunctionMaxLength") and promote this style in the documentation
- or rewrite the rule and all others for consistency reasons

Personally I don't have a strong opinion on this topic, I mainly wanted to share my confusion as it could help to improve detekt

**Comment by schalkms** [comment]
> Personally I don't have a strong opinion on this topic, I mainly wanted to share my confusion as it could help to improve detekt

Thanks for providing feedback here.

**Comment by arturbosch** [comment]
>     * enable that one can write `Suppress("naming.FunctionMaxLength") and promote this style in the documentation

This style is already supported and we should really promote it :)

**Comment by schalkms** [comment]
> I beliebe that a rule name should be as expressive as possible and here FunctionNameMaxLength is the clear winner.

Whilst doing this you need to rename the majority of rules in the naming ruleset (for instance VariableMaxLength & VariableMinLength).

**Comment by BraisGabin** [comment]
Do we want to close this? I think that the rename should be done for 2.0

**Comment by schalkms** [comment]
This means that we have to rename the majority of rules in the naming ruleset. I have a bad feeling concerning user acceptance here.

**Comment by TWiStErRob** [comment]
Is pre-adding an alias for these names something that could be done? So in the end it's swapping out an alias rather than a full rename. People could start using the new name? Maybe this issue was from before aliases were introduced?

**Comment by TWiStErRob** [comment]
Huh, this is already done for [min/max](https://github.com/detekt/detekt/pull/4050)

**Comment by BraisGabin** [comment]
Now we also have a way to deprecate rules so maybe this is actionable right now and we don't need to wait for 2.0

**Comment by schalkms** [comment]
Since there is an alias available for this rule, I'd suggest closing this item.

**Comment by TWiStErRob** [comment]
@schalkms I think we should at least swap the names, so the class and the id is the new name and the alias is the old one?

**Comment by marschwar** [comment]
fixed by #6131
